server/datastore/mysql/hosts.go

```diff
@@ -4402,6 +4403,57 @@ WHERE %s`
 	return nil
 }
 
+// Given a host, attempt to associate with a SCIM user.
+func (ds *Datastore) MaybeAssociateHostWithScimUser(ctx context.Context, host *fleet.Host) error {
+	// Validate that host is not nil.
+	if host == nil {
+		return ctxerr.New(ctx, "MaybeAssociateHostWithScimUser: host is nil")
+	}
+
+	// Check for an existing SCIM user association for the host.
+	var existingSCIMUserID uint
+	checkExistingSQL := `SELECT scim_user_id FROM host_scim_user WHERE host_id = ?`
+	err := sqlx.GetContext(ctx, ds.reader(ctx), &existingSCIMUserID, checkExistingSQL, host.ID)
+	if err == nil {
+		// Existing SCIM user association found, nothing to do.
+		// Bail early so that we don't trigger side-effects downstream like resending profiles.
+		return nil
+	} else if !errors.Is(err, sql.ErrNoRows) {
+		return ctxerr.Wrap(ctx, err, "MaybeAssociateHostWithScimUser: check existing SCIM user for host")
+	}
+
+	// Get any existing MDM IdP record for the host.
+	getMDMIDPSQL := `
+SELECT
+    mdm_idp_accounts.uuid,
+	mdm_idp_accounts.username,
+	mdm_idp_accounts.email,
+	mdm_idp_accounts.fullname
+FROM
+	mdm_idp_accounts
+JOIN host_mdm_idp_accounts ON
```

Comment:
Where in the enrollment process of a Linux/Windows host is `host_mdm_idp_accounts` populated?

Reply:
In the MDM SSO callback [here](https://github.com/fleetdm/fleet/blob/e9b8e47e584967ad322099abb430826f5ab422f1/ee/server/service/mdm.go#L949-L956); if it's coming from the "setup experience" end-user auth then we create this association immediately since we have all the info we need (i.e. the host UUID).  For macOS this happens after the MDM enrollment.